### DISCUSSION TITLE
danfojs-node return wrong data rows

### DISCUSSION THREAD

**Discussion by isaacbizteg** [opensource9ja/danfojs#259] [Q&A] [answered]
I have an excel file `.xlsx` (refer **Image 1**) and refer **Image 2** for my source code in reactjs/typescript/electron js. It outputs the wrong results (refer **Image 3**).

Anyone have idea what's wrong? Thanks in advance

**Image 1**

[![enter image description here][1]][1]

**Image 2**

    const dfd = require("danfojs-node")
    
    dfd.read_excel(inputfile)
        .then((df) => {
    
            //prints the first five columns
            df.head().print()
    
            console.log('df', df)
        }).catch(err => {
            console.log('err', err);
        })

**Image 3**

[![enter image description here][2]][2]

  [1]: https://i.stack.imgur.com/RqBoB.png
  [2]: https://i.stack.imgur.com/QhWNN.png

**Comment by risenW** [comment, marked as answer]
@isaacbizteg This issue is probably caused by the excel file you're trying to load. Can you share the excel file so I can try.

**Reply by isaacbizteg** [reply]
Hi thanks for the response. Here's the excel file for your testing:

[test.xlsx](https://github.com/opensource9ja/danfojs/files/6923278/test.xlsx)

**Reply by risenW** [reply]
This now works well in the latest version (0.3.0). We updated our excel parsing library.

**Comment by spleshakov** [comment]
In my opinion it is a bug and wrong documentation, which I reported here https://github.com/opensource9ja/danfojs/issues/260
@isaacbizteg There is a workaround to fix it though, check out the thread
